File: csharp/src/NetUDP6sender.cs
using System;
using System.Text;
using System.Net;
using System.Net.Sockets;

namespace NetUDP6sender
{
    public class Program
    {
        private static readonly string RECEIVER_ADDRESS = "::1";
        private static readonly int RECEIVER_PORT = 10000;

        public static void Main(string[] args)
        {
            Console.WriteLine("UDP IPv6 datagram socket sender");

            try
            {
                Socket socket = new Socket(AddressFamily.InterNetworkV6, SocketType.Dgram, ProtocolType.Udp);
                Console.WriteLine("- socket created");

                string msg;
                if (args.Length > 0)
                    msg = args[0]; // message is the first parameter, for example "stop" to stop the receiver
                else
                {
                    DateTime now = DateTime.Now;
                    msg = string.Format("This is C# message sent at {0} in {1}", now.ToString("d.M.yyyy"), now.ToString("H:mm:ss"));
                }
                Console.WriteLine("|{0}|", msg);

                byte[] buffer = Encoding.ASCII.GetBytes(msg);
                IPAddress receiver_address = IPAddress.Parse(RECEIVER_ADDRESS);
                IPEndPoint receiver = new IPEndPoint(receiver_address, RECEIVER_PORT);

                int sent_length = socket.SendTo(buffer, receiver);
                IPEndPoint local = (IPEndPoint)socket.LocalEndPoint;
                Console.WriteLine("- message {0}B of {1}B sent to [{2}]:{3} on port {4}", sent_length, msg.Length, receiver_address, RECEIVER_PORT, local.Port);

                socket.Close();
                Console.WriteLine("- socket closed");
            }
            catch (Exception ex)
            {
                Console.WriteLine("- {0}", ex.Message);
            }
        }
    }
}

File: python/src/net_udp6_sender.py
import socket
import sys
from datetime import datetime

RECEIVER_ADDRESS = "::1";
RECEIVER_PORT = 10000;

print("UDP IPv6 datagram socket sender")

sendingSocket = socket.socket(family=socket.AF_INET6, type=socket.SOCK_DGRAM)
print("- socket created")

if len(sys.argv) > 1:
    msg = sys.argv[1] # message is the first parameter, for example "stop" to stop the receiver
else:
    now = datetime.now()
    msg = "This is Python message sent at " + now.strftime("%d.%m.%Y") + " in " + now.strftime("%H:%M:%S")
print("|" + msg + "|")

buffer = str.encode(msg)
receiver_address = (RECEIVER_ADDRESS, RECEIVER_PORT)

msg_length = len(msg)
sendingSocket.sendto(buffer, receiver_address)

(local_address, local_port, local_object3, local_object4) = sendingSocket.getsockname()
print("- message " + str(msg_length) + "B sent to [" + RECEIVER_ADDRESS + "]:" + str(RECEIVER_PORT) + " on port " + str(local_port))

sendingSocket.close();
print("- socket closed")
